domain/ledgeraccount.py

```diff
@@ -0,0 +1,40 @@
+from enum import Enum
+from typing import List
+
+from domain.commontypes.types import AccountID, Amount
+from domain.transaction import Transaction
+
+
+class AccountNature(Enum):
+    CREDIT_ACCOUNT = 1
+    DEBIT_ACCOUNT = 2
+
+
+class LedgerAccount:
+    def __init__(self, account_id: AccountID, name: str, account_nature: AccountNature, transactions: List[Transaction]):
+        self.account_id = account_id
+        self._name = name
+        self._account_nature = account_nature
+        self.transactions = transactions
+
+    def get_balance(self) -> Amount:
+        balance = 0.0
+        if self._account_nature == AccountNature.CREDIT_ACCOUNT:
+            for t in self.transactions:
```

Comment:
duplication of for's loop indicate that 
1) The nature must have more behavior
2) That behaviour must operate on a List of Movements of Credit or Debit

Reply:
i reduced into 1 for with 2 conditionals just extracting for to a higher level than ifs.